ssv/value_check.go

```diff
@@ -11,6 +11,29 @@ import (
 	"github.com/ssvlabs/ssv-spec/types"
 )
 
+// BeaconVoteValueCheckF creates a value check function for a beacon vote
+// It filters out the slashable public keys from the shares and creates a value check function
+// that checks if the beacon vote is slashable by the filtered public keys
+// It also does sanity checks on the beacon vote data
+// It returns an error if shares assigned to the duty are missing
+func BeaconVoteValueCheckF(duty *types.CommitteeDuty, signer types.BeaconSigner, beaconNetwork types.BeaconNetwork,
+	availableShares map[phase0.ValidatorIndex]*types.Share) (qbft.ProposedValueCheckF, error) {
+	duties := duty.ValidatorDuties
+	slashablePublicKeys := []types.ShareValidatorPK{}
+	for _, duty := range duties {
+		if duty.Type == types.BNRoleAttester {
```

Comment:
This is just a little suggestion. For me, it feels easier to reason about it in the following way

```suggestion
	slashablePublicKeys := []types.ShareValidatorPK{}
	for _, duty := range duties {
		if duty.Type == types.BNRoleAttester {
```

Reply:
🧐 

excuse my silliness, I can't see a difference

Reply:
how is duty outside the for loop?

Reply:
Ups! Sorry, I misread it.